06_miscellaneous/floss_licenses_slides.md

```diff
@@ -0,0 +1,173 @@
+- You contribute to an existing software and want to understand what this legally means.
+
+---
+
+## Code Without a License
+
+- Code is creative work and, thus, under **exclusive copyright by default**.
+- Copyright owner is typically the creator (or their employer).
+- Nobody else can copy / distribute / modify
+- If code has more than one contributor, all contributors might become this _"nobody"_.
+- **GitHub terms of service**:
+    - Others can view and fork your repository.
+    - Does not imply that they can use / modify / share your code.
+- Still, you do not have to give a license.
+    - Even though not necessary, putting a copyright notice in a prominent place helps clarifying in this case.
+
+---
+
+## Using a License
+
+- A FLOSS license allows reuse of your code while retaining copyright.
+- To choose or changing a license all copyright owners must agree (if not stated otherwise).
+
+---
+
+## Types of Licenses
+
+... but there are sooo many licenses, they even have different versions. :scream:
+
+Focus on big picture: There are mainly **three types**:
+
+- Permissive licenses (not restrictive)
+- Copyleft licenses (very restrictive)
+- Intermediate licenses (complicated, will not cover here)
+
+... and always choose the newest version of a license.
+
+---
+
+## Permissive Licenses
+
+- In general, you can do what you want if you provide the original license and copyright notice with code.
+    - Modifications might be done under different license
+    - Can ship code as part of commercial software
+- Examples: MIT, Apache, or BSD
+- **MIT** is the way to go if you do not yet know where to go.
+    - Short, easy to understand
+    - Easy to change
+
+---
+
+## MIT License
+
+... so simple, we can actually read and mostly understand it:
+
+```
+Copyright (c) [year] [fullname]
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", ...
+```
+
+---
+
+## Copyleft Licenses
+
+- Main example is **GNU GPL**
+- Two main restrictions:
+    - (A) If you modify software and share it (e.g. sell binary), you need to publish the source code of the modification under the same license.
+    - (B) If you use software in larger works (e.g. through interfaces or data structures), restriction (A) holds for larger works as well.
+        - Example: DuMuX is GPLv3. Thus, if you publish a solver using DuMuX, it needs to be GPLv3 again.
+- **GNU LGPL**: "L" stands for lesser (sometimes also for library)
```

Comment:
Just for me: LGPL is what preCICE does. So businesses can include it in their software as long as they do not change the code. Is that correct?

Reply:
Yes and if they modify preCICE, they only have to publish their version of preCICE openly, not the their complete software.